app/src/main/java/uk/co/neuralcubes/neuralates/ControlFragment.java

```diff
@@ -254,27 +254,22 @@ public void run() {
 
     @Override
     public void updateRobots(final List<Robot> robots) {
-        this.getActivity().runOnUiThread(new Runnable() {
```

Comment:
I might not understand entirely android's model, but this might be called from a different thread (not the ui) shouldn't it be run through `runOnUiThread`?